### PR TITLE
solution

### PR DESCRIPTION
**Pull request by kq333**

### CODE REVIEW

**Review by Amsko6 — APPROVED** [review]

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,16 +1,68 @@
...
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
```
Comment (kasiakubisiak) [inline src/sortStudents.ts:14]:
```suggestion
  Name = 'name',
  Surname = 'surname',
  Age = 'age',
  Married = 'married',
  AverageGrade = 'averageGrade',
```

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,16 +1,68 @@
...
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyArray: Student[] = [...students];
```
Comment (kasiakubisiak) [inline src/sortStudents.ts:24]:
```suggestion
  const studentsArrayCopy: Student[] = [...students];
```
be more descriptive

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,16 +1,68 @@
...
+  switch (sortBy) {
+    case SortType.Name: {
+      copyArray.sort((a, b) => a.name.localeCompare(b.name));
+      break;
+    }
+
+    case SortType.Surname: {
+      copyArray.sort((a, b) => a.surname.localeCompare(b.surname));
+      break;
+    }
```
Comment (kasiakubisiak) [inline src/sortStudents.ts:35]:
```suggestion
    case SortType.Name:
    case SortType.Surname:
      copyArray.sort((firstStudent: Student, secondStudent: Student) => (
        firstStudent[sortBy].localeCompare(secondStudent[sortBy])
      ));
      break;
```
checklist pt 3 - [CODE STYLE] - DRY. Combine logic for sorting by fields with same data types.

File: src/sortStudents.ts [hunk 4]
```diff
@@ -1,16 +1,68 @@
...
+
+    case SortType.Age: {
+      copyArray.sort((a, b) => b.age - a.age);
```
Comment (kasiakubisiak) [inline src/sortStudents.ts:38]:
```suggestion
      copyArray.sort((firstStudent, secondStudent) => secondStudent.age - firstStudent.age);
```
change namings in all code like this :)

File: src/sortStudents.ts [hunk 5]
```diff
@@ -1,16 +1,68 @@
...
+
+    case SortType.Married: {
+      copyArray.sort((a, b) => Number(b.married) - Number(a.married));
+      break;
+    }
+
+    case SortType.AverageGrade: {
+      const avgSort = (a: Student, b: Student):Student[] => {
+        const avgA = a.grades
+          .reduce((acc, grade) => acc + grade, 0) / a.grades.length;
```
Comment (kasiakubisiak) [inline src/sortStudents.ts:50]: also, you could make a function to calculate average grades :)

**Review by kasiakubisiak — CHANGES_REQUESTED** [review]

File: src/sortStudents.ts (hunk 5 above)
Comment (kasiakubisiak) [reply src/sortStudents.ts:50]: Plus use better naming for `avgSort` and `avgA`

**Review by d-wasilewski — APPROVED** [review]

**Pull request closed without merging**